torch/distributed/_fsdp/wrap.py

```diff
@@ -293,7 +303,7 @@ def recursive_wrap(module: nn.Module, auto_wrap_policy: Optional[Callable], **kw
             # decide if we need to wrap the current module,
             # since the left over parameters exceed the number of params to wrap
             remainder = num_params - total_wrapped_params
-            if auto_wrap_policy(module=module, recurse=False, unwrapped_params=remainder):
+            if not only_wrap_children and auto_wrap_policy(module=module, recurse=False, unwrapped_params=remainder):
```

Comment:
recursive_wrap by default would wrap the passed in module as well, but we already called FSDP(model, policy) on it, it will cause issues if we do not skip it like this